### PR TITLE
Do not overwrite identical files

### PR DESCRIPTION
**Pull request by kefirbandi**
If _render_allowed identifies identical source and target, then only returns True if the path refers to a directory, not a file.

### CODE REVIEW

**Comment by sisp** [comment]
Just a test is missing (see https://github.com/copier-org/copier/discussions/1092#discussioncomment-9027925).

Resolves #1092.

**Comment by yajo** [comment]
Cool! Do you know how to add the test then?

**Comment by kefirbandi** [comment]
> Cool! Do you know how to add the test then?

I would use the testcase: [test_skip_if_exists](https://github.com/copier-org/copier/blob/master/tests/test_copy.py#L285) as an example, and write something along the lines of:
```
def test_timestamp_identical(tmp_path: Path) -> None:
    copier.run_copy(str(Path("tests", "demo_skip_dst")), tmp_path)
    modification_time_before = ...
    sleep(2)
    copier.run_copy(str(Path("tests", "demo_skip_dst")), tmp_path)
    modification_time_after = ...

    assert modification_time_before == modification_time_after
   
```
Does it make sense?

**Comment by yajo** [comment]
Ok

File: tests/test_copy.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import os
 import filecmp
```
Comment (sisp) [inline tests/test_copy.py:4]: Ruff is unhappy because of the import order:

```suggestion
import filecmp
import os
```

**Review by sisp — CHANGES_REQUESTED** [review]
Just a minor linter error. LGTM otherwise! :tada:

**Review by yajo — APPROVED** [review]

File: tests/test_copy.py [hunk 2]
```diff
@@ -297,6 +299,16 @@ def test_skip_if_exists(tmp_path: Path) -> None:
...
+def test_timestamp_identical(tmp_path: Path) -> None:
+    copier.run_copy(str(Path("tests", "demo_skip_dst")), tmp_path)
+    modification_time_before = os.path.getmtime(tmp_path / "a.noeof.txt")
+    sleep(2)
+    copier.run_copy(str(Path("tests", "demo_skip_dst")), tmp_path)
+    modification_time_after = os.path.getmtime(tmp_path / "a.noeof.txt")
+
+    assert modification_time_before == modification_time_after
+    
+
```
Comment (sisp) [inline tests/test_copy.py:311]: There are still minor linter errors:

```
> editorconfig-checker.....................................................Failed
> - hook id: editorconfig-checker
> - exit code: 1
>
> tests/test_copy.py:
>    310: Trailing whitespace
>       311: Trailing whitespace
> 
> 2 errors found
```

**Review by sisp — CHANGES_REQUESTED** [review]

**Review by sisp — APPROVED** [review]

**Pull request merged**